### PR TITLE
[AC-5619] Add React

### PR DESCRIPTION
**Pull request by mc-nathan**
To test:
- Run `make run-server`

Should startup a webserver and give you a local URL you can check out.  The local URL should just say "Hello, world".

- Run `make test`

Should successfully run a single test.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,10 +29,27 @@ code-check:
 	@echo eslint not enabled yet
 
 setup: $(GIT_HOOKS_TARGETS)
+	@brew install node
```
Comment (jonkiparsky) [inline Makefile]: homebrew is a mac-specific installer. Is this really the best way to do this?

File: bin/setup [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+set -e
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+NO_COLOR='\033[0m'
+
+# check for node
+if ! command -v node > /dev/null; then
+  printf "💀${RED}   You must install node on your system before setup can continue${NO_COLOR}\n"
```
Comment (jonkiparsky) [inline bin/setup]: Do we really need the emojis and color?

File: Makefile (hunk 1 above)
Comment (jonkiparsky) [reply Makefile]: Also, this setup configuration conflicted with an existing install of node, I had to do a bunch of manual deletion and permissions twiddling to resolve it. We should probably try to make that safe.

**Review by jonkiparsky — CHANGES_REQUESTED** [review]
yarn.lock is a generated file, should it really be committed to the repo?

**Comment by mc-nathan** [comment]
Yes, yarn.lock should be checked in.  It is a record of the resolution of the version closure that has been installed and presumably tested.  This is the file that should determine which versions actually get installed on production.  This is similar to the standard we want to get to in Python, pyenv, and Ruby's bundler.

File: bin/setup (hunk 2 above)
Comment (mc-nathan) [reply bin/setup]: This was a script from thoughtbot and it is essentially the same as the one on in the directory repo.  We should find a way to share it, but I don't think we should edit it on this PR.

File: Makefile (hunk 1 above)
Comment (mc-nathan) [reply Makefile]: OK, I think I made things safer.  A bit tricky to test without starting from a clean box, but it definite avoids messing with any existing version of node or yarn.  Ultimately this should all get handed by the docker stuff, but we haven't gotten there yet.

File: bin/setup (hunk 2 above)
Comment (mc-nathan) [reply bin/setup]: Ended up not needing with the clean up to the Makefile do I deleted it.

**Comment by jonkiparsky** [comment]
Is this error expected? 
screen shot 2018-06-05 at 1 47 42 pm

**Review by jonkiparsky — CHANGES_REQUESTED** [review]
When I load, I see the hello world text, but also "Error: failed to fetch"
I assume this error is not expected?

**Comment by jonkiparsky** [comment]
Closing per discussion with @mc-nathan

**Pull request closed without merging**